src/main/scala/tv/teads/wiremock/extension/FreeMarkerRenderer.scala

```diff
@@ -79,4 +83,44 @@ class FreeMarkerRenderer extends ResponseDefinitionTransformer {
       hash
   }
 
+  class FindFirstStringInArray(requestBody: JsonNode) extends TemplateMethodModelEx {
+    import freemarker.template.TemplateModelException
+
+    @tailrec
+    private def findChildNode(parent: JsonNode, fullPath: String): JsonNode = {
+      val childNodes = fullPath.split('.')
+      if (childNodes.length == 1)
+        parent.findPath(fullPath)
+      else findChildNode(parent.findPath(childNodes.head), childNodes.tail.mkString("."))
+    }
+
+    private def extractPathAndValueFromCondition(filteredChildCondition: String): (String, String) = {
+      val splitFilteredChildCondition = filteredChildCondition.split("==").map(_.trim)
+      if (splitFilteredChildCondition.length != 2) throw new TemplateModelException("Filtered child condition should be like this : car.color == red")
+      (splitFilteredChildCondition(0), splitFilteredChildCondition(1))
+    }
+
+    override def exec(arguments: util.List[_]): SimpleScalar = {
+      if (arguments.size() != 3) {
+        throw new TemplateModelException("Wrong arguments : 3 expected : array node, filtered child condition, wanted node )")
+      }
+      arguments.asScala.toList match {
+        case List(a1: SimpleScalar, a2: SimpleScalar, a3: SimpleScalar) ⇒
+          val (array, filteredChildCondition, wantedChildPath) = (a1.getAsString, a2.getAsString, a3.getAsString)
+          val arrayNode = requestBody.findPath(array)
+          if (!arrayNode.isArray) throw new TemplateModelException("First arg should be an array node")
+          val (filteredChildPath, filteredChildValue) = extractPathAndValueFromCondition(filteredChildCondition)
```

Comment:
This code deserves more space to make it more readable. Use some newline maybe ?